### DISCUSSION TITLE
car docker build

### DISCUSSION THREAD

**Discussion by frestone** [carla-simulator/carla#5608] [Q&A]
Hello
    I built my own carla docker image according to the guide(https://carla.readthedocs.io/en/latest/build_docker_unreal),  the problem is the size of docker image is so large,the size of carla-prerequistes image is **99GB**,and the carla image size is **165GB**,but the official carlasim/carla:latest image is only **16.6GB**
 Does anyone has meet the same problem? 
  
Kind regards

**Comment by throx66** [comment]
Hey frestone, 
I think you the docker instructions still give you a sort of development platform. For slimming things further down, I think we can follow the [non-docker instructions](https://carla.readthedocs.io/en/latest/build_linux/) and make a package for distribution by `make package`.
That will create a new subfolder Dist . The package ready for distribution contains a new DockerFile which can be used to build the final Docker image for deployment, is my understanding.

**Reply by ll7** [reply]
I think this is correct. There are two approaches for using carla in docker. 

1) Build Unreal Engine and Carla from source in docker instead of your host system. -> very large
2) Move an existing carla binary (or package) in a docker environment where it can be executed. -> ~17GB
